php-plugin/src/main/java/fr/greencodeinitiative/php/checks/AvoidGettingSizeCollectionInForLoopCheck.java

```diff
@@ -0,0 +1,71 @@
+package fr.greencodeinitiative.php.checks;
+
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.plugins.php.api.tree.SeparatedList;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.ParenthesisedExpressionTree;
+import org.sonar.plugins.php.api.tree.statement.DoWhileStatementTree;
+import org.sonar.plugins.php.api.tree.statement.ForStatementTree;
+import org.sonar.plugins.php.api.tree.statement.StatementTree;
+import org.sonar.plugins.php.api.tree.statement.WhileStatementTree;
+import org.sonar.plugins.php.api.visitors.PHPSubscriptionCheck;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+
+@Rule(
+        key = AvoidGettingSizeCollectionInForLoopCheck.RULE_KEY,
+        name = AvoidGettingSizeCollectionInForLoopCheck.ERROR_MESSAGE,
+        description = AvoidGettingSizeCollectionInForLoopCheck.ERROR_MESSAGE,
+        priority = Priority.MINOR,
+        tags = {"eco-design", "ecocode", "bad-practice", "performance"}
+)
+public class AvoidGettingSizeCollectionInForLoopCheck extends PHPSubscriptionCheck {
```

Comment:
please change the name of Class it deals with all kind of loops and note only "for" loops.
please do the same with all files :(